File: Backend/airport-backend/src/main/java/vn/iuhcm/airport/luggage/Luggage.java
package vn.iuhcm.airport.luggage;

import java.math.BigDecimal;

public class Luggage {

    public String luggageID; // Primary Key: LG00001
    public BigDecimal weight; // DECIMAL(5,2)
    public String status; // 'Checked-In','Loaded','Unloaded','Missing'
    public String ticketID; // Foreign Key to Ticket

    // Constructor rỗng (Jackson cần)
    public Luggage() {
    }

    // Constructor đầy đủ
    public Luggage(String luggageID, BigDecimal weight, String status, String ticketID) {
        this.luggageID = luggageID;
        this.weight = weight;
        this.status = status;
        this.ticketID = ticketID;
    }
}

File: Frontend/airport-frontend/app/types/Luggage.ts
// File: app/types/Luggage.ts

export type LuggageStatus = 
  | 'Checked-In' 
  | 'Loaded' 
  | 'Unloaded' 
  | 'Missing';

export interface Luggage {
  /** Primary Key: LG00001 */
  luggageID: string;

  /** Weight (DECIMAL(5,2) in DB, mapped to number) */
  weight: number;

  /** Status must be one of the defined LuggageStatus types */
  status: LuggageStatus;

  /** Foreign Key to Ticket */
  ticketID: string;
}